extensions/bitwarden/src/search.tsx

```diff
@@ -10,45 +10,65 @@ import {
   Toast,
   Clipboard,
   Action,
+  LocalStorage,
+  popToRoot,
 } from "@raycast/api";
-import { Item, VaultStatus } from "./types";
+import { Item } from "./types";
 import React, { useEffect, useState } from "react";
-import treeify from "treeify";
-import { filterNullishPropertiesFromObject, codeBlock, titleCase, faviconUrl } from "./utils";
-import { useBitwarden } from "./hooks";
-import { TroubleshootingGuide, UnlockForm } from "./components";
+import { codeBlock, titleCase, faviconUrl } from "./utils";
 import { Bitwarden } from "./api";
+import { SESSION_KEY } from "./const";
+import { TroubleshootingGuide, UnlockForm } from "./components";
+import { existsSync } from "fs";
 
 const { fetchFavicons, primaryAction } = getPreferenceValues();
 
-export default function Search(): JSX.Element {
-  try {
-    const bitwardenApi = new Bitwarden();
-    const [state, setSessionToken] = useBitwarden(bitwardenApi);
+function useSession() {
+  const [state, setState] = useState<{ isLoading: boolean; token?: string }>({ isLoading: true });
 
-    if (state.vaultStatus === "locked") {
-      return <UnlockForm setSessionToken={setSessionToken} bitwardenApi={bitwardenApi} />;
-    }
-    return <ItemList bitwardenApi={bitwardenApi} sessionToken={state.sessionToken} vaultStatus={state.vaultStatus} />;
-  } catch (error) {
+  useEffect(() => {
+    LocalStorage.getItem<string>(SESSION_KEY).then((token) => setState({ isLoading: false, token }));
+  }, []);
+
+  return {
+    token: state.token,
+    active: !state.isLoading,
+    setToken: async (token: string) => {
+      await LocalStorage.setItem(SESSION_KEY, token);
+      setState({ isLoading: false, token });
+    },
+    deleteToken: async () => {
+      await LocalStorage.removeItem(SESSION_KEY);
+      setState({ isLoading: false });
+    },
+  };
+}
+
+export default function Search() {
+  const {
+    cliPath = process.arch == "arm64" ? "/opt/homebrew/bin/bw" : "/usr/local/bin/bw",
+    clientId,
+    clientSecret,
+  } = getPreferenceValues();
```

Comment:
```suggestion
  const { cliPath: cliPathPreference, clientId, clientSecret } = getPreferenceValues();
  const cliPath = cliPathPreference || process.arch == "arm64" ? "/opt/homebrew/bin/bw" : "/usr/local/bin/bw";
```

This destructuring doesn't handle when the preference is set to an empty string - i.e. been set and then removed in Raycast UI.

Reply:
Good catch !